src/query/functions/src/scalars/array.rs

```diff
@@ -606,4 +613,381 @@ pub fn register(registry: &mut FunctionRegistry) {
             }
         }),
     );
+
+    for left in ALL_SIGNED_INTEGER_TYPES {
+        with_number_mapped_type!(|NUM_TYPE| match left {
+            NumberDataType::NUM_TYPE => {
```

Comment:
can we use `register_factory` and agg combinators?

The DataType could be evaluated as `agg.input_type()`.
Thus we could make the code path cleaner.

Reply:
The `return_type` is related to the type of the aggregate operation, and the type of the operation needs to be obtained by the second argument, so the `return_type` cannot be known in `FunctionSignature`.